tests/iterators_tests/test_balanced_serial_iterator.py

```diff
@@ -38,6 +54,12 @@ def _test_index_iterator_no_shuffle():
     assert indices3[0] == index_list[1]
     assert indices3[1] == index_list[2]
 
```

Comment:
I think it would be better to separate serialization tests from `_test_index_iterator_no_shuffle` to make each do one thing. IMO, it is acceptable that this change occurs code duplication to some extent. For example how about making `test_index_iterator_serialization` that runs `_test_index_iterator_serialization_no_shuffle` and `_test_index_iterator_serialization_with_shuffle` (function names are tentative).